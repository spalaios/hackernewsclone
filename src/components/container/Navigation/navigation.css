.hacker-news-name {
  font-weight: 900;
  font-size: 0.875rem;
}
.navigation-bar {
  background: #fb693f;
}
.hacker-news-logo {
  border: 1px solid #fff;
}

.navigation__tabs p {
  font-size: 0.8333rem;
}
