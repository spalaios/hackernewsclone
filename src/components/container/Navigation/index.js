import React from 'react';
import './navigation.css';

function navigateTo(route, history) {
  history.push(route);
}

class NavigationBar extends React.Component { // TODO: implement from nav bar based 
  constructor(props) {
    super(props);
    this.state = {
      new: 0,
    };
  }

  componentDidMount() {
    // this.getCurrentPathName();
  }

  componentWillUnmount() {
    // kill all request
  }

  navigateTo = (route) => {
    if (route === 'logo') {
      route = '';
    }
    this.props.history.push(`/${route}`);
  }

  getCurrentPathName = () => { // TODO: find out why this functioon is not working directly in render 
    return this.props.history.location.pathname.split('/')[1];
  }

  render() {
    return (
      <section className="navigation d-flex border navigation-bar">
        <div className="navigation__hn-logo mx-1 d-flex align-self-center cursor" onClick={() => this.navigateTo('logo')} >
          <img
            src={'/src/assets/images/hnlogo.gif'}
            alt="Y"
            className="hacker-news-logo"
          />
        </div>
        <div className="navigation__hn-name d-flex align-items-center mx-1 cursor">
          <p className="text-capitalize hacker-news-name mb-0">Hacker News</p>
        </div>
        <div className="navigation__tabs d-flex">
          <p className={`mb-0 mx-1 cursor ${(this.props.history.location.pathname.split('/')[1] === 'newest') ? 'text-white' : ''}`} onClick={() => this.navigateTo('newest') } >new <span className="text-black">|</span></p>
          <p className={`mb-0 mx-1 cursor ${(this.props.history.location.pathname.split('/')[1] === 'comments') ? 'text-white' : ''}`} onClick={() => this.navigateTo('comments') }>comments <span className="text-black">|</span></p>
          <p className={`mb-0 mx-1 cursor ${(this.props.history.location.pathname.split('/')[1] === 'show') ? 'text-white' : ''}`} onClick={() => this.navigateTo('show') }>show <span className="text-black">|</span></p>
          <p className={`mb-0 mx-1 cursor ${(this.props.history.location.pathname.split('/')[1] === 'ask') ? 'text-white' : ''}`} onClick={() => this.navigateTo('ask') }>ask <span className="text-black">|</span></p>
          <p className={`mb-0 mx-1 cursor ${(this.props.history.location.pathname.split('/')[1] === 'jobs') ? 'text-white' : ''}`} onClick={() => this.navigateTo('jobs') }>jobs</p>
        </div>
      </section>
    );
  }
}

export default NavigationBar;
