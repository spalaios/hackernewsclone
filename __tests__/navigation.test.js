// import React from 'react';
// import renderer from 'react-test-renderer';
// import Navigation from '../src/components/container/Navigation/index';

// test('Testing the navigation compoent', () => {
//   const navigationComponent = renderer.create(
//     <Navigation />,
//   );
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from '../src/components/container/Navigation/index';
// import MoreButton from '../../src/components/presentation/Button/Button';

test('Testing the navigation component', () => {
  const component = renderer.create(
    <Navigation />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});