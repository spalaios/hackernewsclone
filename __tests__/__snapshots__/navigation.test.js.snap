// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the navigation component 1`] = `
<section
  className="navigation d-flex border navigation-bar"
>
  <div
    className="navigation__hn-logo mx-1 d-flex align-self-center cursor"
  >
    <img
      alt="Y"
      className="hacker-news-logo"
      src="/src/assets/images/hnlogo.gif"
    />
  </div>
  <div
    className="navigation__hn-name d-flex align-items-center mx-1 cursor"
  >
    <p
      className="text-capitalize hacker-news-name mb-0"
    >
      Hacker News
    </p>
  </div>
  <div
    className="navigation__tabs d-flex"
  >
    <p
      className="mb-0 mx-1 cursor"
    >
      new |
    </p>
    <p
      className="mb-0 mx-1 cursor"
    >
      comments |
    </p>
    <p
      className="mb-0 mx-1 cursor"
    >
      show |
    </p>
    <p
      className="mb-0 mx-1 cursor"
    >
      ask |
    </p>
    <p
      className="mb-0 mx-1 cursor"
    >
      jobs
    </p>
  </div>
</section>
`;
